src/miner.h

```diff
@@ -160,7 +160,7 @@ class BlockAssembler
 public:
     BlockAssembler(const CChainParams& chainparams);
     /** Construct a new block template with coinbase to scriptPubKeyIn */
-    CBlockTemplate* CreateNewBlock(const CScript& scriptPubKeyIn);
+    std::unique_ptr<CBlockTemplate> CreateNewBlock(const CScript& scriptPubKeyIn);
```

Comment:
But does this mean we want all functions that currently return a bare pointer that will become caller ownership to returning `unique_ptr`?
This is probably the kind of c++11 refactoring that we want to leave for later.